addons/account/models/account_move_line.py

```diff
@@ -3082,7 +3082,7 @@ def _get_installments_data(self, payment_currency=None, payment_date=None, next_
 
             # Early payment discount.
             # In that case, we want to report the difference of the epd and display it on the UI.
-            if move._is_eligible_for_early_payment_discount(payment_currency or line.currency_id, payment_date):
+            if move.state != 'cancel' and move._is_eligible_for_early_payment_discount(payment_currency or line.currency_id, payment_date):
```

Comment:
I think you could move this check directly into `_is_eligible_for_early_payment_discount` (I don't think a canceled move should ever be eligible for early payment discount and same thought from CHKL)

Reply:
> I think you could move this check directly into `_is_eligible_for_early_payment_discount` (I don't think a canceled move should ever be eligible for early payment discount and same thought from CHKL)

@proose The code has been updated can you please check? Thanks :)

Reply:
Yup, looks good to me, you can open the PR if you think you did everything